test/e2e/tests/dapp-interactions/block-explorer.spec.js

```diff
@@ -38,19 +37,17 @@ describe('Block Explorer', function () {
         await driver.clickElement({ text: 'View on explorer', tag: 'p' });
 
         // Switch to block explorer
-        await driver.waitUntilXWindowHandles(2);
-        const windowHandles = await driver.getAllWindowHandles();
-        await driver.switchToWindowWithTitle('E2E Test Page', windowHandles);
-        const body = await driver.findElement(
-          '[data-testid="empty-page-body"]',
-        );
+        await driver.switchToWindowWithTitle('E2E Test Page');
 
         // Verify block explorer
-        assert.equal(await body.getText(), 'Empty page by MetaMask');
-        assert.equal(
-          await driver.getCurrentUrl(),
-          'https://etherscan.io/address/0x5CfE73b6021E818B776b421B1c4Db2474086a7e1',
-        );
```

Comment:
this is an anti-pattern: we should wait for the value we want, instead of grabbing the selector and then asserting it has the correct value, as it can be the case the value is not yet loaded